sygnal/notifications.py

```diff
@@ -14,32 +14,33 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import typing
-from typing import Any, Dict, List, Optional
 
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union
+
+from opentracing import Span
 from prometheus_client import Counter
 
 from .exceptions import InvalidNotificationException, NotificationDispatchException
 
-if typing.TYPE_CHECKING:
+if TYPE_CHECKING:
     from .sygnal import Sygnal
 
 
 class Tweaks:
-    def __init__(self, raw):
-        self.sound = None
+    def __init__(self, raw) -> None:
+        self.sound: Optional[str] = None
 
         if "sound" in raw:
             self.sound = raw["sound"]
 
 
 class Device:
     def __init__(self, raw):
-        self.app_id = None
-        self.pushkey = None
-        self.pushkey_ts = 0
-        self.data = None
-        self.tweaks = None
+        self.app_id: Optional[str] = None
+        self.pushkey: Optional[str] = None
+        self.pushkey_ts: float = 0
```

Comment:
do we know if this is float or int?
OK fine I'll stop being lazy and I'll look it up...

https://spec.matrix.org/unstable/push-gateway-api/#post_matrixpushv1notify
 says integer :)

(would you be able to validate that the number we pull out of the raw dict is an int before you shove it in this field?)

I think this kind of dance will do it, making use of `.get`'s defaults...:
```python
raw_pushkey_ts = raw.get("pushkey_ts", 0)
if not isinstance(raw_pushkey_ts, int):
    raise InvalidNotificationException("Device with non-integral pushkey_ts")
self.pushkey_ts: int = raw_pushkey_ts
```